### DISCUSSION TITLE
How to write a conditional function in MATLAB?

### DISCUSSION THREAD

**Discussion by npn0010** [casadi/casadi#3615] [Q&A] [answered]
How can I write the conditional function $f(x)= 0,~x < 0;~0.5,~0 \leq x \leq 1;~1,~x > 1$ in MATLAB using `conditional`? I have been searching online for awhile and it's not quite clear to me how to do this. Thanks in advance

**Comment by sandeep026** [comment, marked as answer]
Use if_else().

$$
u(t)=\left\{\begin{array}{ll}
-\frac{2}{3 l}\left(1-\frac{t}{3 l}\right), & 0 \leq t \leq 3 l \\
0, & 3 l \leq t \leq 1-3 l \\
-\frac{2}{3 l}\left(1-\frac{1-t}{3 l}\right), & 1-3 l \leq t \leq 1
\end{array}
$$




```
t=cs.SX.sym('t',1)
f=cs.Function('f',[t],[cs.if_else(t<=3*l,-(2/(3*l))*(1-1/(3*l)*t),cs.if_else(t>=1-3*l,-(2/(3*l))*(1-(1-t)/(3*l)),0))])
```

**Reply by npn0010** [reply]
Ok I see. So are nested if_else's the only way to do it even when there's many conditions?

**Reply by jaeandersson** [reply]
You can do casadi.Function.conditional. That one is short-circuiting (i.e. only evaluates a specific branch) and supports multiple conditions. But the syntax is a bit more cumbersome - you need to create separate functions for each case.
